Bounding box generated by add dataset flow can't be indexed

**Describe the bug**
If you create a new dataset with a bounding box roughly around Australia (e.g. use an ASIC Business Names CSV), it will generate a bounding box that can't be indexed by the indexer because it isn't using the right coordinates:

```
[ERROR] [08/16/2019 04:32:51.447] [indexer-akka.actor.default-dispatcher-32] [akka.actor.ActorSystemImpl(indexer)] Failed to convert bounding box to spatial data for dataset magda-ds-1f9ca335-c976-4fa0-84e8-a84e0d4b6a79: Invalid latitude (y) range: 114.594962. Should be within -90 to +90
```

**To Reproduce**
1. Go to `/dataset/add`
2. Create a new dataset, selecting "Australia" as the spatial extent
3. `kubectl logs -f` the indexer and wait
4. You'll see that the dataset can't be indexed.

**Expected behavior**
- The add dataset flow should generate valid coordinates
- Better yet, the indexer should be able to understand these coordinates, given that we see them a lot in other datasets.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -26,6 +26,7 @@ Indexer:
 -   Fixed indexer throws an error when temporalCoverage aspects intervals is an empty array
 -   Index datasets with tenant ID.
 -   Fixed indexer throws an error when affiliatedOrganisation field is created
+-   Fixed indexer incorrect parsing bounding box data in spatialCoverage aspect
 
 Cataloging:
 

--- a/magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala
+++ b/magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala
@@ -264,11 +264,13 @@ trait RegistryConverters extends RegistryProtocols {
                 None
               } else {
                 Some(
+                  // --- see magda-registry-aspects/spatial-coverage.schema.json
+                  // --- Bounding box in order minlon (west), minlat (south), maxlon (east), maxlat (north)
                   BoundingBox(
                     bbox.elements(3).convertTo[Double],
                     bbox.elements(2).convertTo[Double],
-                    bbox.elements(0).convertTo[Double],
-                    bbox.elements(1).convertTo[Double]
+                    bbox.elements(1).convertTo[Double],
+                    bbox.elements(0).convertTo[Double]
                   )
                 ).map(Location(_))
               }

